docs/kvrocks-search.md

```diff
@@ -0,0 +1,269 @@
+(query_expr)              |
+query_expr AND query_expr |
+query_expr OR query_expr  |
+NOT query_expr            |
+tag_field HASTAG tag      |
+num_atom NUM_OP num_atom  |
+vec_field <-> vec < range
+```
+
+where the numeric operation `NUM_OP` can be:
+
+```
+< | <= | > | >= | !=
+```
+
+and the `num_atom` can be:
+
+```
+num_field | num_literal
+```
+
+Also, these literals inside the query in can be parameters `@param_name`, 
+e.g. `a < 233` can be `a < @num` with `PARAMS 1 num 233` supplied to the `FT.SEARCHSQL`.
+
+## RediSearch query syntax
+
+Currently Kvrocks also supports a subset of [the RediSearch query syntax](https://redis.io/docs/latest/develop/interact/search-and-query/advanced-concepts/query_syntax/).
+
+RediSearch controls the evolution of the query syntax through [dialect versioning](https://redis.io/docs/latest/develop/interact/search-and-query/advanced-concepts/dialects/).
+Currently, Kvrocks supports `DIALECT 2`.
+And in future developments, we may support higher versions of dialect (currently, 3 and 4), but `DIALECT 1` is NOT considered for support.
+
+The followings are the query clauses currently supported in Kvrocks, and you can compose them via `clause | clause` (OR), `clause clause` (AND) and `-clause` (NOT):
+- `*`, i.e. `true` in SQL;
+- `@num_field:[NUM_BOUND NUM_BOUND]`, e.g. `@a:[1 (3]` means `a >= 1 and a < 3`;
+- `@tag_field:{tag [|tag ...]}`, e.g. `@b:{x | y}` means `b hastag x or b hastag y`;
+- `@vec_field:[VECTOR_RANGE range $vec]` for vector range query.
+
+where `NUM_BOUND` can be:
+```
+  num   
+| (num
+| INF
+| +INF
+| -INF
+```
+
+Also KNN query without prefiltering is supported:
+```
+* => [KNN n @vec_field $vec]
+```
+
+Also, these literals inside the query in can be parameters `$param_name`, 
+e.g. `@a:[inf 233]` can be `@a:[inf $num]` with `PARAMS 1 num 233` supplied to the `FT.SEARCH`.
+
+## Field types
+
+An index in RediSearch consists of multiple fields, and fields can be in different types.
+Currently, Kvrocks supports three field types:
+- `TAG`: a tag field can hold a set of string tags, to filter rows by specific tags in queries;
+- `NUMERIC`: a numeric field can hold a floating point number;
+- `VECTOR`: a vector field can hold a vector, for performing vector search.
+
+### Tag
+
+Field properties:
+```
+SCHEMA field_name TAG
+  [SEPARATOR sep]
+  [CASESENSITIVE]
+```
+
+By default, the `SEPARATOR` is `,` and `CASESENSITIVE` is not set.
+
+The only operation for tag field in queries is to check if a row is labeled by tag, i.e. `tag_field HASTAG tag` in SQL.
+
+### Numeric
+
+Numeric field has no field properties, i.e.
+```
+SCHEMA field_name NUMERIC
+```
+
+As shown in the query syntax, numeric fields can be used in numeric comparison to filter data.
+
+### Vector
+
+Field properties:
+```
+SCHEMA field_name VECTOR HNSW nargs
+  TYPE FLOAT64
+  DIM dim
+  DISTANCE_METRIC L2 | IP | COSINE
+  [M m]
+  [EF_CONSTRUCTION ef_construcion]
+  [EF_RUNTIME ef_runtime]
+  [EPSILON epsilon]
```

Comment:
Do we want to add a comment in the end saying that properties wrapped by`[...]` in schemas mean optional args?